From e6ab44aa45f34743452993d41480ccddd5db0caa Mon Sep 17 00:00:00 2001
From: qbisi <qbisicwate@gmail.com>
Date: Sun, 22 Jun 2025 08:43:55 +0800
Subject: [PATCH] medwrapper: update med version expected

Signed-off-by: qbisi <qbisicwate@gmail.com>
---
 src/MEDWrapper/MED_Factory.cxx | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/MEDWrapper/MED_Factory.cxx b/src/MEDWrapper/MED_Factory.cxx
index a9da2e570..06aaaaaa2 100644
--- a/src/MEDWrapper/MED_Factory.cxx
+++ b/src/MEDWrapper/MED_Factory.cxx
@@ -56,15 +56,15 @@ extern "C"
 // Empty file content for MED-4.x should be provided (EMPTY_FILE_4x) and method CreateEmptyMEDFile should be updated.
 // *******************************************************************************************************************
 
-#define MED_MAJOR_EXPECTED 4
-#define MED_MINOR_EXPECTED 1
+#define MED_MAJOR_EXPECTED 5
+#define MED_MINOR_EXPECTED 0
 #if MED_MAJOR_NUM != MED_MAJOR_EXPECTED
   #error "MED major version does not correspond to the expected version, fix the minor and major compatibility values in CheckCompatibility method (MED_VERSIONS_APPEND_COMPATIBLE) and set the correct expected version (MED_MAJOR_EXPECTED, MED_MINOR_EXPECTED)"
 #endif
 #if MED_MINOR_NUM != MED_MINOR_EXPECTED
   #error "MED minor version does not correspond to the expected version, fix the minor and major compatibility values in CheckCompatibility method (MED_VERSIONS_APPEND_COMPATIBLE) and set the correct expected version above (MED_MAJOR_EXPECTED, MED_MINOR_EXPECTED)"
 #endif
-#define MED_VERSIONS_APPEND_COMPATIBLE {41, 40, 32, 33} // --- 10*major + minor (the 3rd digit, release, is not used here,
+#define MED_VERSIONS_APPEND_COMPATIBLE {50, 41, 40, 32, 33} // --- 10*major + minor (the 3rd digit, release, is not used here,
                                                         //                       med uses always the latest available)
                                                         // --- The first in the list should be the default: current version
 
-- 
2.49.0

