diff --git a/src/DSC/DSC_User/Datastream/Calcium/CalciumC.c b/src/DSC/DSC_User/Datastream/Calcium/CalciumC.c
index 5fdc9f47..f76448e8 100644
--- a/src/DSC/DSC_User/Datastream/Calcium/CalciumC.c
+++ b/src/DSC/DSC_User/Datastream/Calcium/CalciumC.c
@@ -27,6 +27,7 @@
 #include "calcium.h"
 #include "calciumf.h"
 #include "CalciumFortranInt.h"
+#include "CalciumMacroCInterface.h"
 #include <stdlib.h>
 #include <stdio.h>
 #include <string.h>
@@ -42,7 +43,9 @@ InfoType ecp_fint_ (void * component, char* nomVar, float t);
 InfoType ecp_fini_ (void * component, char* nomVar, int i);
 InfoType ecp_efft_ (void * component, char* nomVar, float t);
 InfoType ecp_effi_ (void * component, char* nomVar, int i);
-
+//Following have been added for declaration
+InfoType ecp_fin_ (void * component, int code);
+InfoType ecp_cd_ (void * component, char* instanceName);
 /************************************/
 /* INTERFACES DE LECTURE EN 0 COPIE */
 /************************************/
@@ -54,6 +57,17 @@ InfoType ecp_effi_ (void * component, char* nomVar, int i);
 /* Attention en cas de lectures multiples : le buffer retourné est le même */
 /* Attention si les niveaux sont actifs le buffer peut être supprimé automatiquement par calcium. */
 
+CALCIUM_C2CPP_INTERFACE_C_(intc,int,int,float,)
+CALCIUM_C2CPP_INTERFACE_C_(long,long,long,float,)
+CALCIUM_C2CPP_INTERFACE_C_(integer,integer,cal_int,float,)
+CALCIUM_C2CPP_INTERFACE_C_(int2integer,integer,int,float,)
+CALCIUM_C2CPP_INTERFACE_C_(long2integer,integer, long,float,)
+CALCIUM_C2CPP_INTERFACE_C_(float,float,float,float, )
+CALCIUM_C2CPP_INTERFACE_C_(double,double,double,double,)
+CALCIUM_C2CPP_INTERFACE_C_(float2double,double,float,float, )
+CALCIUM_C2CPP_INTERFACE_C_(bool,bool,int,float,)
+CALCIUM_C2CPP_INTERFACE_C_(cplx,cplx,float,float,)
+CALCIUM_C2CPP_INTERFACE_C_(str,str,char*,float,)
 #define CALCIUM_EXT_LECT_INTERFACE_C_(_name,_timeType,_type,_typeName,_qual) \
   InfoType ecp_##_name (void * component, int mode,                     \
                         _timeType * ti, _timeType * tf, int * i,        \
diff --git a/src/DSC/DSC_User/Datastream/Calcium/CalciumMacroCInterface.h b/src/DSC/DSC_User/Datastream/Calcium/CalciumMacroCInterface.h
new file mode 100644
index 00000000..6e402317
--- /dev/null
+++ b/src/DSC/DSC_User/Datastream/Calcium/CalciumMacroCInterface.h
@@ -0,0 +1,18 @@
+#define _CALCIUM_MACRO_C_INTERFACE_H_
+
+#define CALCIUM_C2CPP_INTERFACE_C_(_name,_porttype,_type,_timeType,_qual)        \
+  InfoType ecp_lecture_##_name (void * component, int dependencyType,            \
+                                        _timeType * ti,                          \
+                                        _timeType * tf, long * i,                \
+                                        const char * const nomvar, size_t bufferLength,    \
+                                        size_t * nRead, _type _qual ** data );   \
+                                                                                 \
+                                                                                 \
+  void ecp_lecture_##_name##_free ( _type _qual * data);                         \
+                                                                                 \
+                                                                                 \
+  InfoType ecp_ecriture_##_name (void * component, int dependencyType,           \
+                                        _timeType *t,                            \
+                                        long  i,                                 \
+                                        const char * const nomvar, size_t bufferLength,     \
+                                        _type _qual * data );                    \
