diff --git a/pyproject.toml b/pyproject.toml
index 7f572acaa..bd681adeb 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -63,7 +63,6 @@ firedrake-clean = "firedrake.scripts.firedrake_clean:main"
 firedrake-preprocess-bibtex = "firedrake.scripts.firedrake_preprocess_bibtex:main"
 firedrake-status = "firedrake.scripts.firedrake_status:main"
 pyop2-clean = "pyop2.compilation:clear_compiler_disk_cache"
-spydump = "pyop2.scripts.spydump:main"
 
 
 [project.optional-dependencies]
@@ -151,6 +150,7 @@ build-backend = "setuptools.build_meta"
 script-files = [
   "firedrake/scripts/firedrake-zenodo",
   "scripts/firedrake-run-split-tests",
+  "pyop2/scripts/spydump",
 ]
 
 [tool.setuptools.package-data]
